lib/src/widgets/adw/switch.dart

```diff
@@ -0,0 +1,675 @@
+// Copyright 2014 The Flutter Authors => BSD-3 License
```

Comment:
I believe this code is taken from either another library or flutter Switch themselves. Is there a reason to import the class as a Adw class instead of using the generic version and changing it's theme through `yaru` or `adwaita`?

Reply:
I dont think there is such colors for checkbox in the current adwaita theme or yaru theme.

Reply:
I don't think I was very clear with my comment sorry :S.

I have two questions :).

1. Could this be replaced by a simple theming of the generic material switch widget? `adwaita` theme can have a switchTheme section that would enable to correctly color the element through the theme instead of manually linking the class to Adwaita Colors. Again I'm asking this to try to remove the color logic out of libadwaita as much as possible :).

2. If not, could we, in this class, extract the colors for the [switchTheme](https://api.flutter.dev/flutter/material/SwitchTheme-class.html) instead of making this class dependent on adwaita colors?

I don't the original code used as inspiration for the this class so it's really up to you.
Overall though, it would be nice to remove the need to use AdwColors defined in `libadwaita` and usually rely on the basic `ThemeData` flutter class

Reply:
@MalcolmMielle Will you update the switchTheme for adwaita package, I will surely update the code once that happens.

Reply:
I have resolved this by using switchTheme by default and fallback values if  they are null.